### PR TITLE
Fix/Chore [vercel integration] improved form validation, combobox component upgrade, loading states added

### PR DESCRIPTION
**Pull request by MildTomato**
## What kind of change does this PR introduce?

- combo box added
- validation improvements for install form
- loading states added into combobox triggers
- disabled project linker submit button when form not complete

## What is the current behavior?

Please link any relevant issues here.

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Add any other context or screenshots.

Combo box added
Screenshot 2023-07-20 at 4 02 44 PM

Validation improvements
Screenshot 2023-07-20 at 4 12 55 PM

Loading states
Screenshot 2023-07-20 at 4 12 33 PM

### CODE REVIEW

File: studio/components/interfaces/Integrations/ProjectLinker.tsx [hunk 1]
```diff
@@ -170,42 +242,88 @@ const ProjectLinker = ({
...
+                    <CommandEmpty_Shadcn_>No results found.</CommandEmpty_Shadcn_>
+                    <CommandGroup_Shadcn_>
+                      {filteredForeignProjects.map((project) => {
+                        return (
+                          <CommandItem_Shadcn_
+                            value={project.id}
+                            key={project.id}
+                            className="flex gap-2 items-center"
+                            onSelect={(id) => {
+                              console.log('id set', id)
```
Comment (joshenlim) [inline studio/components/interfaces/Integrations/ProjectLinker.tsx:303]: sneaky console log

File: studio/pages/integrations/vercel/install.tsx [hunk 2]
```diff
@@ -48,42 +72,53 @@ const VercelIntegration: NextPageWithLayout = () => {
...
-          router.query.organizationSlug = firstOrg.slug
-        }
-      },
+  const { data: organizationsData, isLoading: isLoadingOrganizationsQuery } =
+    useOrganizationsQuery()
+
+  useEffect(() => {
+    if (organizationsData !== undefined && integrationData !== undefined) {
+      const firstOrg = organizationsWithInstalls(organizationsData)?.[0]
+      console.log('firstOrg', firstOrg)
```
Comment (joshenlim) [inline studio/pages/integrations/vercel/install.tsx:81]: sneaky console.log

File: studio/pages/integrations/vercel/install.tsx [hunk 3]
```diff
@@ -168,55 +203,85 @@ const VercelIntegration: NextPageWithLayout = () => {
...
   const dataLoading = isLoadingVercelIntegrationCreateMutation || isLoadingOrganizationsQuery
 
+  const disableInstallationForm =
+    (isLoadingVercelIntegrationCreateMutation && !dataLoading) ||
+    // disables installation button if integration is already installed and it is Marketplace flow
+    (selectedOrg && selectedOrg?.installationInstalled && source === 'marketplace' && !dataLoading)
+      ? true
+      : false
+
+  console.log('selectedOrg', selectedOrg)
```
Comment (joshenlim) [inline studio/pages/integrations/vercel/install.tsx:213]: sneaky console.log

File: studio/pages/integrations/vercel/install.tsx [hunk 4]
```diff
@@ -168,55 +203,85 @@ const VercelIntegration: NextPageWithLayout = () => {
...
+            <ScaffoldContainer className="flex flex-col gap-6 grow py-8">
+              <ScaffoldColumn className="mx-auto w-full max-w-md">
+                <h1 className="text-xl text-scale-1200">Choose organization</h1>
+                <>
+                  <Markdown content={`Choose the Supabase organization you wish to install in`} />
+                  <OrganizationPicker
+                    integrationName="Vercel"
+                    organizationsWithInstalledData={organizationsWithInstalledData}
+                    onSelectedOrgChange={(e: string) => {
+                      console.log('e:', e)
```
Comment (joshenlim) [inline studio/pages/integrations/vercel/install.tsx:230]: sneakyyy

```suggestion
```

File: studio/pages/integrations/vercel/install.tsx [hunk 5]
```diff
@@ -168,55 +203,85 @@ const VercelIntegration: NextPageWithLayout = () => {
...
+
+                      const org = organizationsWithInstalledData?.find((org) => org.slug === e)
+                      console.log('const:', org)
```
Comment (joshenlim) [inline studio/pages/integrations/vercel/install.tsx:233]: sneakyyyy

```suggestion
```

**Comment by MildTomato** [comment]
sorry, forgot to clean up @joshenlim

**Review by alaister — APPROVED** [review]
LGTM!

**Comment by MildTomato** [comment]
issue with syncing env vars

File: studio/components/interfaces/Integrations/ProjectLinker.tsx [hunk 6]
```diff
@@ -61,21 +80,40 @@ const ProjectLinker = ({
...
+    : undefined
+
+  const selectedVercelProject = vercelProjectId
+    ? filteredForeignProjects.find((x) => x.id?.toLowerCase() === vercelProjectId?.toLowerCase())
+    : undefined
+
+  function onCreateConnections() {
+    const projectDetails = selectedVercelProject
+
+    console.log('projectDetails', projectDetails)
```
Comment (alaister) [inline studio/components/interfaces/Integrations/ProjectLinker.tsx:104]: cheeky console.log

File: studio/.gitignore [hunk 7]
```diff
@@ -33,3 +33,5 @@ yarn-error.log*
 .env.production.local
 
 .vercel
+
+/public/dashboard
```
Comment (alaister) [inline studio/.gitignore:37]: ?

Comment (MildTomato) [reply studio/.gitignore:37]: for fonts

File: studio/components/interfaces/Integrations/ProjectLinker.tsx (hunk 6 above)
Comment (MildTomato) [reply studio/components/interfaces/Integrations/ProjectLinker.tsx]: fixed

**Review by alaister — APPROVED** [review]
Thanks Jonny!

**Pull request merged**